src/jit/target.h

```diff
@@ -531,6 +532,7 @@ typedef unsigned char   regNumberSmall;
   #define FEATURE_FASTTAILCALL     1       // Tail calls made as epilog+jmp
   #define FEATURE_TAILCALL_OPT     1       // opportunistic Tail calls (i.e. without ".tail" prefix) made as fast tail calls.
   #define FEATURE_SET_FLAGS        0       // Set to true to force the JIT to mark the trees with GTF_SET_FLAGS when the flags need to be set
+  #define MAX_PASS_SINGLEREG_BYTES      8  // Maximum size of a struct passed in a single register (double).
```

Comment:
Do we have `MAX_RETURN_SINGLEREG_BYTES`? `x64` specifies that `__m128` is returned in register, but passed by ref for the default calling convention.

Reply:
We don't have that yet, though presumably it (or something like it) will be added in subsequent ABI cleanups.

Reply:
Is fixing the x64 ABI for the new vector primtivies on the slate for 3.0?
* Note: I am talking about the x64 default calling convention; not the currently unsupported `__vectorcall` convention

I think the important bits are:
* __m64 types, are passed as if they were integers of the same size.
  * Even if we don't support any intrinsic methods for `Vector64<T>` on the managed side, we should still pass/return it properly.
* For these aggregate types passed as a pointer, including __m128, the caller-allocated temporary memory must be 16-byte aligned.
* A scalar return value that can fit into 64 bits is returned through RAX; this includes __m64 types.
* Non-scalar types including floats, doubles, and vector types such as __m128, __m128i, __m128d are returned in XMM0.

Reply:
Short answer: I hadn't planned any changes to the x64 ABI:
- I hadn't realized that `__m64` types were passed in floating point registers - is that type supported/specified for the linux ABI?
- Where does the 16-byte alignment requirement come from? Is that in the ABI spec?

> Non-scalar types including floats, doubles, and vector types such as __m128, __m128i, __m128d are returned in XMM0.

Is this supported without `__vectorcall`?

Reply:
> is that type supported/specified for the linux ABI

The above are all from the Windows x64 calling convention defined here: https://docs.microsoft.com/en-us/cpp/build/x64-calling-convention?view=vs-2019.

The `__m64`, `__m128`, and `__m256` types are also defined by the Linux "System V" ABI (both 32-bit and 64-bit), but I haven't double checked the requirements for that yet...

> Is this supported without __vectorcall

Yes, as per the above, this is in the default Windows x64 calling convention; so it should be applicable to what we support in the JIT today.

Reply:
Also, for reference. I am prototyping the ability to marshal blittable generics here: https://github.com/dotnet/coreclr/pull/23899

The Windows x64 failures can be seen in the latest commit and it would likely be useful to validate that the ARM/ARM64 tests pass on that PR after this PR is merged.